File: tests/fixtures/swift_sample/Greeter.swift
import Foundation

protocol Named {
    var name: String { get }
}

class Greeter: Named {
    let name: String

    init(name: String) {
        self.name = name
    }

    func greet() -> String {
        return formatMessage(name)
    }
}

func formatMessage(_ name: String) -> String {
    return "hello, \(name)"
}

File: tests/fixtures/python_sample/greeter.py
"""A tiny greeting library used by the static-analysis fixtures."""


class Greeter:
    """Holds a name and produces greetings."""

    def __init__(self, name: str) -> None:
        self.name = name

    def greet(self) -> str:
        """Return a greeting for the stored name."""
        return format_message(self.name)


def format_message(name: str) -> str:
    """Build a greeting string for *name*."""
    return f"hello, {name}"
